main.cpp

```diff
@@ -77,22 +77,38 @@ void scan_demo(WiFiInterface *wifi)
 void http_demo(NetworkInterface *net)
 {
     TCPSocket socket;
+    nsapi_error_t response;
 
     printf("Sending HTTP request to www.arm.com...\r\n");
 
     // Open a socket on the network interface, and create a TCP connection to www.arm.com
     socket.open(net);
-    socket.connect("www.arm.com", 80);
+    response = socket.connect("www.arm.com", 80);
+    if(0 != response) {
+        printf("Error connecting: %d\r\n", response);
+        socket.close();
+        return;
+    }
 
     // Send a simple http request
     char sbuffer[] = "GET / HTTP/1.1\r\nHost: www.arm.com\r\n\r\n";
-    int scount = socket.send(sbuffer, sizeof sbuffer);
-    printf("sent %d [%.*s]\r\n", scount, strstr(sbuffer, "\r\n")-sbuffer, sbuffer);
+    response = socket.send(sbuffer, sizeof sbuffer);
+    if(response < 0) {
```

Comment:
If we are fixing error checking we should also be making sure the whole buffer is transmitted, as socket send doesn't guarantee all the bytes are actually transmitted.